os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -257,88 +257,77 @@ static void up_dumpstate(void)
 	uint32_t istacksize;
 #endif
 
-	/* Get the limits on the user stack memory */
+	/* Get the limits for each type of stack */
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
-
 #if CONFIG_ARCH_INTERRUPTSTACK > 3
-	/* Get the limits on the interrupt stack memory */
-
 	istackbase = (uint32_t)&g_intstackbase;
 	istacksize = (CONFIG_ARCH_INTERRUPTSTACK & ~3);
-
-	/* Show interrupt stack info */
-
-	lldbg("sp:     %08x\n", sp);
-	lldbg("IRQ stack:\n");
-	lldbg("  base: %08x\n", istackbase);
-	lldbg("  size: %08x\n", istacksize);
-#ifdef CONFIG_STACK_COLORATION
-	lldbg("  used: %08x\n", up_check_intstack());
 #endif
 
-	/* Does the current stack pointer lie within the interrupt
-	 * stack?
-	 */
-
-	if (sp <= istackbase && sp > istackbase - istacksize) {
-		/* Yes.. dump the interrupt stack */
+	/* Check if the assert location is in user thread or IRQ handler */
 
-		up_stackdump(sp, istackbase);
+	if (g_irq_nums[2]) {
+		/* Assert in nested irq */
+		lldbg("Code asserted in nested IRQ state!\n");
+	} else if (g_irq_nums[1]) {
+		/* Assert in irq */
+		lldbg("Code asserted in IRQ state!\n");
+	} else {
+		/* Assert in user thread */
+		lldbg("Code asserted in normal thread!\n");
 	}
 
-	/* Extract the user stack pointer if we are in an interrupt handler.
-	 * If we are not in an interrupt handler.  Then sp is the user stack
-	 * pointer (and the above range check should have failed).
-	 */
+	/* Print IRQ handler details if required */
 
-	if (current_regs) {
-		sp = current_regs[REG_R13];
-		lldbg("sp:     %08x\n", sp);
-	}
-
-	lldbg("User stack:\n");
-	lldbg("  base: %08x\n", ustackbase);
-	lldbg("  size: %08x\n", ustacksize);
-#ifdef CONFIG_STACK_COLORATION
-	lldbg("  used: %08x\n", up_check_tcbstack(rtcb));
+	if (g_irq_nums[1]) {
```

Comment:
Do you need to check it again? Until this line, you check this twice. I don't know why it needs to be checked again.